File: app/src/main/java/com/maciejkozlowski/comunicationshowcaseapplication/MainFragment.java
package com.maciejkozlowski.comunicationshowcaseapplication;

import android.content.Context;

import com.maciejkozlowski.fragmentutils.FragmentUtils;

import androidx.fragment.app.Fragment;

/**
 * Created by Maciej Kozłowski on 09.04.2018.
 */
public class MainFragment extends Fragment {

    private ReplaceFragmentListener replaceFragmentListener;

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        replaceFragmentListener = FragmentUtils.getListener(this, ReplaceFragmentListener.class);
    }
}

File: app/src/main/java/com/maciejkozlowski/comunicationshowcaseapplication/MainFragment.kt
package com.maciejkozlowski.comunicationshowcaseapplication

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.fragment_main.*

/**
 * Created by Maciej Kozłowski on 07.04.2018.
 */
class MainFragment : Fragment() {
    
    companion object {
        fun newInstance() = MainFragment()
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_main, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        button.setOnClickListener { onButtonClicked() }
    }

    private fun onButtonClicked() {
        (activity as MainActivity).showDifferentFragent()
    }
}